src/Geometry.cpp

```diff
@@ -1941,7 +1935,7 @@ void Geometry::segmentize2D(Track* track, double z_coord) {
 
     /* Add the segment to the Track */
     track->addSegment(new_segment);
-
+    delete new_segment;
```

Comment:
Good catch. Since _segments.push_back(*segment) uses *segment which is an instance and an automatic copy will be made.